web/src/features/Dashboard/components/Dashboard.tsx

```diff
@@ -620,8 +620,46 @@ const Dashboard = (props: Props) => {
     );
   };
 
-  const { data: selectedAppClusterDashboardResponse } =
-    useSelectedAppClusterDashboardWithIntercept({ refetchInterval: 2000 });
+  const {
+    data: selectedAppClusterDashboardResponse,
+    isSlowLoading: isSlowLoadingSelectedAppClusterDashboard,
+  } = useSelectedAppClusterDashboardWithIntercept({ refetchInterval: 2000 });
+
+  const { isSlowLoading: isSlowLoadingNextAppVersion } =
+    useNextAppVersionWithIntercept();
+
+  console.log(state.slowLoader);
+  // show slow loader if any of the apis are slow loading
+  useEffect(() => {
+    // since this is new and we may need to debug it, leaving these in for now
+    // console.log("isSlowLoadingLicense", isSlowLoadingLicense);
+    // console.log("isSlowLoadingNextAppVersion", isSlowLoadingNextAppVersion);
+    // console.log(
+    //   "isSlowLoadingSelectedAppClusterDashboard",
+    //   isSlowLoadingSelectedAppClusterDashboard
+    // );
+    if (
+      !state.slowLoader &&
+      (isSlowLoadingLicense ||
+        isSlowLoadingNextAppVersion ||
+        isSlowLoadingSelectedAppClusterDashboard)
+    ) {
+      setState({ slowLoader: true });
+      return;
+    }
+    if (
+      state.slowLoader &&
+      !isSlowLoadingLicense &&
+      !isSlowLoadingNextAppVersion &&
+      !isSlowLoadingSelectedAppClusterDashboard
+    ) {
+      setState({ slowLoader: false });
+    }
+  }, [
+    isSlowLoadingLicense,
+    isSlowLoadingNextAppVersion,
+    isSlowLoadingSelectedAppClusterDashboard,
+  ]);
```

Comment:
This effect fires if any of the apis slowLoading state changes. We will eventually refactor this `useEffect` out and use the loading state returned from the hooks directly. We need to do this in several places so we'll just do them all once. Using `useEffect` is fine but it adds unnecessary complexity.